--- echo_client_v6_naif.py	2016-10-16 01:59:34.449953405 +0200
+++ echo_client_v6.py	2016-10-16 02:00:56.867059224 +0200
@@ -3,8 +3,16 @@
 
 HOST = 'daring.cwi.nl'    # The remote host
 PORT = 50007              # The same port as used by the server
-with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as s:
-    s.connect((HOST, PORT))
-    s.sendall(b'Hello, world')
-    data = s.recv(1024)
+
+res = socket.getaddrinfo(HOST, PORT, socket.AF_INET6,
+                         socket.SOCK_STREAM, 0, socket.AI_V4MAPPED)
+for af, socktype, proto, canonname, sa in res:
+    try:
+        with socket.socket(af, socktype, proto) as s:
+            s.connect((HOST, PORT))
+            s.sendall(b'Hello, world')
+            data = s.recv(1024)
+            break
+    except socket.error as msg:
+        continue
 print('Received', repr(data))
